src/Faulkner/avoidance.py

```diff
@@ -5,65 +5,81 @@
 import loghelper
 
 class avoidance:
-    def __init__(self, state_vars, params):
+    def __init__(self, state_vars, params, history):
         self.state_vars = state_vars
         self.params = params
+        self.history = history
     
     #def is_obstructed(self):
     #    return self.front() or self.fleft() or self.frontAndFright() or self.frontAndFleft() or self.frontAndFleftAndFright()
         #return self.front() or self.frontAndFright() or self.frontAndFleft() or self.frontAndFleftAndFright()
 
-    def is_obstructed(self):
-        return self.state_vars.regions['front'] <= self.params.avoid_distance or self.state_vars.regions['fleft'] <= self.params.avoid_distance or self.state_vars.regions['fright'] <= self.params.avoid_distance
+    #def is_obstructed(self):
+    #    return self.state_vars.regions['front'] <= self.params.avoid_distance or self.state_vars.regions['fleft'] <= self.params.avoid_distance or self.state_vars.regions['fright'] <= self.params.avoid_distance
+    def obstacle_near_centre(self, regions):
+        return regions['front'] <= self.params.avoid_distance or regions['fleft'] <= self.params.avoid_distance or regions['fright'] <= self.params.avoid_distance
     
+    def is_obstructed(self, regions):
+        self.regions = regions
+        history_window = self.history.getLastN(self.params.is_obstructed_window)
```

Comment:
I'm not sure what "N" refers to in "getLastN".

Reply:
I'll change that to read getLastNIterations

Reply:
changed